cumulusci/robotframework/Salesforce.py

```diff
@@ -24,9 +25,25 @@
 lex_locators = {}  # will be initialized when Salesforce is instantiated
 
 
-class RandomStringGenerator:
+class StringGenerator:
+    def __init__(self, func):
+        self.func = func
+
     def __str__(self):
-        return String().generate_random_string()
+        return self.func()
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
+
+
+class FakerTemplateLibrary:
+    faker = faker.Faker()
+
+    def __getattr__(self, name):
+        # defer loading heavy faker library until actually needed
+        return StringGenerator(
+            lambda *args, **kwargs: self.faker.format(name, *args, **kwargs)
+        )
```

Comment:
Can you add docstrings for these classes and move them to a utils module?

Reply:
Done.